### ISSUE TITLE
when app is called resign active method or is entered background, app is crashed

### ISSUE THREAD

**Issue by kaannesinn** [mixpanel/mixpanel-iphone#741]
**Integration Method:** CocoaPods 
**Xcode Version:**  xcode 8.3.3
**Library Version:** mixpanel 3.1.5
**Platform:** iOS 
**Language:** / Objective-C
**Description:** 

Hi, 

when the application is entered background or is called resign active by user opens applications on bckground  with home screen button or by showing any native popup and get call appwillresignactive method, the application crashes. 

we think there could be a bug on the library while doing these cases. appwillresignactive method is called with an observer and app crashes with these logs:

2017-10-31 11:08:16.579 hopi2[3930:393970] Encountered an uncaught exception. All Mixpanel instances were archived.
2017-10-31 11:08:16.580 hopi2[3930:393970] Debug details follow:
-[__NSCFString appWillResignActive:]: unrecognized selector sent to instance 0x175663640
(
    "4   hopi2                               0x0000000100596c4c GADMurmurHash3 + 772",
    "5   CoreFoundation                      0x00000001862105ec <redacted> + 692",
    "6   libobjc.A.dylib                     0x0000000197ccc3b4 <redacted> + 116",
    "7   libc++abi.dylib                     0x00000001974c9bb4 <redacted> + 16",
    "8   libc++abi.dylib                     0x00000001974c973c __cxa_rethrow + 144"
)
2017-10-31 11:08:16.580 hopi2[3930:393970] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFString appWillResignActive:]: unrecognized selector sent to instance 0x175663640'
*** First throw call stack:
(0x18621022c 0x197ccc0e4 0x1862172f4 0x1862140a8 0x18611696c 0x1861b6010 0x1860f30f0 0x187022a80 0x18acf48a8 0x18af213ec 0x18af179f0 0x18e9093c8 0x1861c7fc8 0x1861c70d0 0x1861c56f4 0x1860f0f74 0x18fa0b6fc 0x18acf2d94 0x1002df660 0x198376a08)
libc++abi.dylib: terminating with uncaught exception of type NSExceptionu

**Expected Behavior:** 
application has to be continue to work properly. it has to send mixpanel events after calling appwillresignactive method. 

could there be any problems on this version ? do we have to update the mixpanel library?

**Comment by pchien** [comment]
Hi @kaannesinn. Are you sure the crash is inside the Mixpanel library? We archive Mixpanel instances and track crashes which will write to the log, but that doesn't mean that we caused the crash.

**Comment by zihejia** [comment]
Hi @kaannesinn, we will close this one for now, feel free to reopen it if the problem still exits.

**Comment by kaannesinn** [comment]
Hi @zihejia, we didnt understand what the reason was for this crash. We suspected for 2 libraries on our application. One of them is Mixpanel and the other is 3rd party library for location tracking. But after updating the new version of Mixpanel this crash was gone. I checkouted the commit before this version, crash didnt exist.

Thanks @pchien for your answer also. Our company talked Mixpanel support and tech teams and they said that processing a main thread process on background thread could cause this bug. And with new version now app doesnt crash. (we have solved it nearly 1-1.5 months ago)
